<!DOCTYPE html><html lang="en"><head><script async src="https://www.googletagmanager.com/gtag/js?id=UA-114897551-4"></script><script>window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'UA-114897551-4');
</script><script type="text/javascript" src="//platform-api.sharethis.com/js/sharethis.js#property=5ac2443d1fff98001395ab6c&amp;product=sticky-share-buttons" async="async"></script><title>Java Video Tutorial 6 | Coder Coacher - Coaching Coders</title><meta content="Java Video Tutorial 6 - All technical stuff in one place" name="description"><meta name="keywords" content="education, coding, programming, technology, nodejs, mongodb, software, computer science, engineering, teaching, coaching, coder, learning, java, kotlin, machine learning, AI, ML, tech talks, angular, javascript, js, typescript"><meta name="viewport" content="width=device-width, initial-scale=1.0"><link rel="stylesheet" href="/css/font.css"><link rel="stylesheet" href="/css/bootstrap.css"><link rel="stylesheet" href="/css/style.css"><link rel="stylesheet" href="/css/coder-coacher.css"></head><body><div class="container-fluid"><h1 class="site-title"><a href="/">Coder Coacher</a></h1><hr><h4 class="site-subtitle text-right">Coaching Coders</h4></div><div id="amzn-assoc-ad-99d6751e-2392-4004-ad16-73aa8385d9d0"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=99d6751e-2392-4004-ad16-73aa8385d9d0"></script><div class="post__breadcrumb"><div class="container"><ol class="breadcrumb"><li><a href="/">Coder Coacher</a></li><li><a href="/Derek-Banas/">Derek Banas</a></li><li class="active">â¤µ</li></ol></div></div><h2 class="post__title"><b>Java Video Tutorial 6</b></h2><h5 class="post__date">2012-01-06</h5><div class="container"><div class="video-responsive"><iframe width="560" height="315" src="https://www.youtube.com/embed/EWj60p8esD0" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe></div><div class="post__text">well hello internet and welcome to part
6 of my Java video tutorial today we're
going to talk about exceptions and
exception to handling and an exception
is just an object that is created
whenever an error occurs and the
exception itself tells you what error
specifically occurred and we're going to
go through some code here to explain
everything to you there's mainly two
types of exceptions you have those
located in Java Lang runtime exception
and these are exceptions that are thrown
during the normal course of the program
just running and for the most part they
are your responsibility or actually they
are your responsibility in totality you
have to catch these then you have and if
you don't understand what I'm doing here
don't worry about it
java.lang exceptions and what these are
are exceptions that are checked by the
compiler and if you do not protect
against these errors the compiler will
warn you and will not allow you to run
your program so we're going to see
examples these though if that doesn't
make any sense to you and we have our
common exceptions these are what you're
most likely to come across do you have
arithmetic exception and this just
occurs whenever an arithmetic operation
goes wrong like a division by zero or
something like that and another common
one is class not found exception and
it's just called whenever a class is
called that doesn't exist and you have a
legal argument exception in an illegal
argument exception is called whenever a
method is passed an illegal argument
very very common then you have index out
of bounds
this might not make sense right now
because we haven't covered arrays I'm
just letting you know what's coming out
there and it's just thrown whatever you
try to access and index in an array or a
string that doesn't exist and if that
doesn't make sense you don't worry about
it then you have input mismatch
exception and this is called whenever if
you're using the scanner method that
we've been using here lately it occurs
whenever you try to input the wrong data
type and then you have i/o exception
which is also very common if you're
doing i/o related activities inside of
java and this is thrown whenever any
type of i/o exception occurs and I'm
going to give you this very specific
example of that so let's just start here
creating our class and it's going to be
called public class Java lesson 6 just
continuing off of what we did before an
N public static void
and I trust that you know what these
mean otherwise see the previous
tutorials because I've gone over this a
bunch of times and here is our main
program area where we're going to be
calling most of our code so we're just
going to start off we're going to dive
right into it so I'm going to do a
divided i0 I'm going to call this
function and it's just going to pass
value of two and I think you know what's
going to be coming up here and if not
you will so we're going to create our
class inside here public static void
public meaning it's available to
everybody static means that a class can
call it a and void means it does not
return anything it's going to accept an
integer a and then this is where we are
going to try to catch an error catch an
exception what you have to do is put it
between braces so you just put try
inside of here and then inside of this
little try block we're going to do
something that is may trigger an error
in our situation is definitely going to
trigger
print line we're going to take whatever
the value of a is and divided by zero
and I guarantee you that is going to
trigger an error and after you do this
this is where all the potential problem
code exists and then you use catch and
you're going to in this situation we're
going to try to catch the problem that
we think is going to occur and you just
type in arithmetic exception like we
talked about before followed by the
letter E and that is going to be where
the exception object is placed then
inside of there if you want to print out
different things or let's just say you
want to print on a custom error message
if somebody tries to divide by zero or
in this situation this doesn't have
anything really to do with the division
by zero it's just you're trying to
perform some sort of arithmetic that
cannot be done well you can just put a
custom message in here that says you
can't do that I don't know you can put
anything in there you want and of course
don't forget to put this little dot
inside of here so we can print this out
to screen and of course make that
lowercase and if we run this guy you can
see over here that our custom message
printed out on the screen so let's just
see what would happen if we didn't have
this so let's get rid of that all
together so we're not going to catch the
error this time and let's run it again
this is what we call runtime error
meaning that it's going to let me run
the program and it's going to print on
an error message and you can see right
there exactly what that error message is
and it's basically saying it divided by
zero error occurred and you cannot do
this okay so that's bad so let's just
change our code back to the way that it
was all right now we're back to
something it'll actually work but you
can also aside from just
turning this error message which is
custom Java has built-in error messages
that you can print that are kind of
family friendly or just don't look like
the whole world's crashing down on you
if you want to use one of them you just
type in e get message and a little
braces there and that's going to print
out a nicer message onto the screen
actually it's not going to really tell
you that much but it can give you a
little bit more exact than that mess
division by zero that is the error in
your code and then you could also print
the exception name and an error message
and all you need to do there is just go
system.out.print line and go e dot to
string and there's two braces there
semicolon at the end and you can see
here on the third line is printing out
the actual exception name and then it
has the little message there at the end
so exception that was triggered is
arithmetic exception and division by
zero is the actual problem and if you
wanted to print out the actual stack
trace or the original error that we saw
before let's just come in here and do it
here delete this guy out I'm just going
to tell you E
print stack trace and if you do that
it's going to just print out screen the
specific error that we saw previously
that's triggered no matter what see
there it is okay so there's a whole
bunch of different ways of printing out
error messages or controlling an
exception now I'm going to show you how
to do a bunch of other things with this
captions pretty much I'm going to cover
everything just like in other tutorials
so in this situation we're going to need
to import a library and we went over how
to do that before in the past and this
one we're going to be using the scanner
object and we're also going to be
looking for a scanner specific exception
and we use scanner to do user input just
like we did previously in the past
tutorials so we're going to type in
static scanner and I'm going to name it
user input remember this can have any
name that you want to give it is equal
to new inner and we're going to say
system in which is the keyboard again
just like I talked about previously and
there we are now we're able to do user
input which is useful and in this
situation we're going to go to system
out dot print line let's just change it
to print so that we don't have space
there and I'm going to say how old are
you that's going to ask the question and
then we're going to create int age is
equal to check valid age is what this
functions going to do for us and then we
need to actually create this so check
valid age we're going to
create it then we're going to go public
static again can be called by a class
int going to return an integer and we'll
just paste that in there and it doesn't
accept an attribute so that's all fine
and dandy then we're going to create our
try block again and here we're going to
return to the person user input next int
it's expecting an integer to be entered
by the user and it just received that
user input now what happens if they do
not enter an integer well we're going to
have to catch that potential error so
we're going to go catch I want to go
input mismatch exception and remember we
created this guy pair of Java utility
well not only is the scanner tool in
there but also this exception that we're
using right here so you definitely need
that to be able to work on and we will
also want to put a little e inside of
there which is the object for the
exception itself and then what we want
to do is we're going to go user input
because if it's in here we want to
pretty much flush the previous user
input if it was negative and by putting
user input next in there that's in
essence what it does skips over what the
last user input whatever the person
typed in its keyboard and it sort of
like skips to the next one so that's
just sort of in there just disregard
previous inputs because we know that the
previous input was wrong so
system.out.print and let's just say that
isn't a whole number and just to be real
simple with that and then in this
situation we're going to return zero
remember we got to return something
because here we have int so it's
expecting that something be return an
integer be returned so then we got to
come up here and we are correct for that
so we're going to jump back into this
guy we're going to say if age is not
equal to zero and then we're gonna go
system out on you our age plus years old
let's run this guy okay it says how old
are my well let's go in here and put a
valid number let's say 19 which my guess
really is invalid but it's a whole
number anyway and you see that it just
printed everything out fine and dandy
but what if somebody comes in here and
instead six-and a-half say that isn't a
whole number but nothing crash nothing
went wrong and that is exactly what
we're using exceptions for or exception
catching so I'm going to show you
something a little bit more complicated
I'm going to use the java io library in
this situation and i
longer need scanners so let's get rid of
it and all this code is available
underneath the video there's a link to
it so you can look at it and it's
heavily commented like there's a lot of
stuff in the code even that I don't even
talk about the tutorial and let's just
come in here and get rid of all that
stuff I'm going to create a new function
and this guy is going to be called get a
file
we haven't handled file i/o and things
like that but we will and this is just a
common error and I just want to go
through it and like I said before if you
don't quite get something that's no big
deal we're gonna cover it over and over
and over again
alright so there's a function we're
going to be calling and it's just called
get a file so here inside of the main
function we're going to call it or since
they get a file and we're going to put a
file in here who does not exist so we're
going to call it some stuff on purpose
all right so there's a semicolon and
this guy's going to call this method
down here so what are we going to do
when you create another try block and
actually what I'm going to do here is
just to show you what happens whenever
you try to use a checked expression and
remember a checked expression is a
expression that the compiler catches so
the compiler says hey you have to check
for this potential problem our potential
problem in this situation is some stuff
text doesn't exist so let's say we go
file input stream and know we have not
covered this yet in previous tutorial so
don't worry about it it's just me trying
to open up a file so file is equal to
new file input stream filename here
we're going to go out there and we're
going to open up this file you're going
to see what happens whenever you try to
execute something that requires that the
exception be checked we're gonna run it
and you can see here errors galore and
what it's saying unhandled exception
type file not found exception so let's
say try write like this this is going to
make the compiler all nice and happy
about everything here and then we're
going to go catch and catch this new guy
and file not found exception e and we're
gonna say system out dot print line and
we're gonna say sorry can't find that
file and then this guy in this situation
is going to run because we caught that
exception that it required us to do and
it just as sorry can't find that file
well another thing is you can also catch
numerous different exceptions you can
have numerous
different catch blocks here so another
common error is IO exception so let's
come in here and let's catch that to
look at IO exception II and in this
situation we're just going to go system
out this is like a general sort of error
occurred and we're just going to say
unknown IO error now you always want on
whenever you create these exceptions you
want to have the most specific one first
and then the least specific one second
and just to show you another one if you
wanted to use a catch-all exceptions
it's real easy to do so this will
actually catch every single exception
but you always want to use this last
because you want to try to solve the
problem rather than just given a vague
answer which in this situation is pretty
much your only option you know you don't
want to print this out to the screen for
the people to see because it looks like
your programs junk that let's just say
some error occurred and that's pretty
much what it means because like it has
no idea what this error is because it
catches every single error all right no
if I ran this the same thing is going to
show up here because this is specific to
the actual error so this is what's going
to print however if we had some other
error currents out of here and nothing
to do with any of this then we might get
down to this point where catch the all
exception block here would be hit so
this catches every exception this
catches a very specific exception this
catches a less exact exception error and
then you also have finally this guy will
run no matter what so if you have a
finally block inside of here it's going
to run whether things turned out well or
not whether an exception was called or
not it's there mainly for cleanup duty
it doesn't necessarily have to do
anything or what is normally is used for
is at the end of your program to close
off any connections you might have with
a database or close any files you had to
open you know basic cleanup duty that's
what finally is used for but just
understand if you use finally it's
definitely going to execute whether an
exception was triggered or not finally
is always called and just to be you know
cover everything here basically let's
come up here in this area right here and
let's say you want to just ignore an
exception if it occurs well it's real
simple just go catch and let's say you
want to ignore class my found exception
and as this tutorial continues I'm going
to show you examples how this works but
if you just wanted to ignore a situation
where our class isn't found you know
there you go you just ignored it because
you did nothing
you caught the error but you did nothing
with it nothing to try to solve it and
another thing you could do this is only
in Java 7 though this isn't and all the
other different javis you can actually
look to catch multiple different
exceptions all at one time and how you
do that's real simple you just put an
orange side of there and go like that so
this is going to catch the class not
found exception and also the IO
exception because that older symbols in
there and it's going to ignore them if
they should come up and then the final
sort of thing you will come across
whenever you're dealing with exceptions
and let's just get rid of this
altogether here is what's called a
thrown exception now if you would want
the main function up here to handle any
exceptions that occur down inside of
this method called get a file what you
need to do is come in here after get a
file actually right here and you go
throws and let's just say you want to
put IO exception as a potential
exception that would throw or would
occur in this or file not found and I
just separated those with a calm inside
of there and you just put that before
the opening bracket now what's going to
happen is if you come down here inside
of get a file and you go file input
stream files equal to new file input
stream file name and you actually try to
execute this what it's going to do is
it's going to hit this point and because
throws IO exception and file not found
exception is here it's actually going to
throw the error back to main and main
needs to be able to handle it so what
does main need in this situation in this
situation I'm covering everything you
would normally do this but I'm just
trying to cover every single thing there
is so here we're gonna go like that and
then in this situation we need to catch
those exceptions so IO exception e
system
I don't print line and I or occurred and
you can see here after I executed that
and there i OE recurred and it didn't
throw an error down inside of here it
allowed me to compile everything and
that was just a way for me of able to
catch those exceptions in main through
the use of throws so there's a whole
bunch of things pretty much everything
you can do with exceptions except for
the most
arcane bizarre things leave any
questions or comments below up next I'm
going to be covering objects on your
programming in Java till next time</div></div><div class="container-fluid bottom-ad"><div id="amzn-assoc-ad-6a809dda-347a-4187-8a86-91faf94575da"></div><script async src="//z-na.amazon-adsystem.com/widgets/onejs?MarketPlace=US&amp;adInstanceId=6a809dda-347a-4187-8a86-91faf94575da"></script></div><div class="text-center">We are a participant in the Amazon Services LLC Associates Program, an affiliate advertising program designed to provide a means for us to earn fees by linking to Amazon.com and affiliated sites.</div><script>(function(w, d){
    var b = d.getElementsByTagName('body')[0];
    var s = d.createElement("script"); s.async = true;
    var v = !("IntersectionObserver" in w) ? "8.6.0" : "10.4.2";
    s.src = "https://cdnjs.cloudflare.com/ajax/libs/vanilla-lazyload/" + v + "/lazyload.min.js";
    w.lazyLoadOptions = {};
    b.appendChild(s);
}(window, document));</script></body></html>